Added AttachThreadInput to User32 namespace.

BarryNolte opened the pull request:
Added the DllImport and api signature to the User32 namespace.

BarryNolte committed:
Added AttachThreadInput to User32 namespace.

vbfox commented on src/User32/User32.cs:
```diff
@@ -246,6 +246,19 @@ public static partial class User32
+        // public static extern bool AttachThreadInput(int idAttach, int idAttachTo, bool fAttach);
+        [DllImport("user32.dll", EntryPoint = "AttachThreadInput")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool AttachThreadInput(uint idAttach, uint idAttachTo, [MarshalAs(UnmanagedType.Bool)] bool fAttach);
```
We don't use `uint` in type signatures prefering to always expose the signed type (so `int` here)

vbfox commented on src/User32/User32.cs:
```diff
@@ -246,6 +246,19 @@ public static partial class User32
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.To get extended error information, call GetLastError. Windows Server2003 and WindowsXP:  There is no extended error information; do not call GetLastError.This behavior changed as of WindowsVista.</returns>
+        // [DllImport(nameof(User32), SetLastError = true)]
+        // public static extern bool AttachThreadInput(int idAttach, int idAttachTo, bool fAttach);
+        [DllImport("user32.dll", EntryPoint = "AttachThreadInput")]
```
`"user32.dll"` should be `nameof(User32)`, no need to specify the `EntryPoint` and the function use `SetLastError` for it's error signaling so it should have `, SetLastError = true`.

vbfox commented:
Thanks for the PR !

Just a few formatting fixed and it's all good for me 😉

vbfox commented on src/User32/User32.cs:
```diff
@@ -246,6 +246,19 @@ public static partial class User32
+        /// <param name="idAttachTo">The identifier of the thread to which idAttach will be attached.This thread cannot be a system thread. A thread cannot attach to itself. Therefore, idAttachTo cannot equal idAttach.</param>
+        /// <param name="fAttach">If this parameter is TRUE, the two threads are attached.If the parameter is FALSE, the threads are detached.</param>
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.To get extended error information, call GetLastError. Windows Server2003 and WindowsXP:  There is no extended error information; do not call GetLastError.This behavior changed as of WindowsVista.</returns>
+        // [DllImport(nameof(User32), SetLastError = true)]
```
That declaration and signature actually looks good :)

If only it wasn't commented

BarryNolte committed:
Changed uint to int for AttachThreadInput

Changed the thread id values from uint to int, making my little c/c++ brain explode.

BarryNolte committed:
Uncommented/deleted code

Used the commented out DllImport and deleted the unneeded delaration.

AArnott commented on src/User32/User32.cs:
```diff
@@ -246,6 +246,16 @@ public static partial class User32
+        /// <param name="idAttach">The identifier of the thread to be attached to another thread.The thread to be attached cannot be a system thread.</param>
+        /// <param name="idAttachTo">The identifier of the thread to which idAttach will be attached.This thread cannot be a system thread. A thread cannot attach to itself. Therefore, idAttachTo cannot equal idAttach.</param>
+        /// <param name="fAttach">If this parameter is TRUE, the two threads are attached.If the parameter is FALSE, the threads are detached.</param>
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.To get extended error information, call GetLastError. Windows Server2003 and WindowsXP:  There is no extended error information; do not call GetLastError.This behavior changed as of WindowsVista.</returns>
```
You are missing several spaces between words and sentences. Likely this happened when you copied and pasted the MSDN docs and C# "formatted" the text by removing spaces after periods. It's important when you copy and paste that after paste, you immediately hit Undo to revert the formatting changes (but keep the paste).

BarryNolte committed:
Fixed formatting of api description

AArnott reviewed (approved):
Awesome. Thank you!
